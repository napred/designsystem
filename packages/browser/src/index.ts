export * from './createStyleApplicator';
export * from './css';
export * from './hooks';
export * from './DesignSystem';
export * from './types';

export { default as css } from './css';
export { default as createComponent } from './createComponent';
export { default as createStyle } from './createStyle';
export { default as createStyleApplicator } from './createStyleApplicator';
export { default as DesignSystem } from './DesignSystem';

export * from '@napred/ds';
