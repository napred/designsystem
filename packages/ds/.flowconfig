/Users/Michal/Work/napred/designsystem/.flowconfig