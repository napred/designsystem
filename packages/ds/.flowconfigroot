/Users/Michal/Work/napred/designsystem/.flowconfigroot