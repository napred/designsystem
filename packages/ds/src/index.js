// @flow

import DesignSystemContext from './context';

export { DesignSystemContext };
export * from './hooks';
export { default as Box } from './Box';
export { default as DesignSystem } from './DesignSystem';
export { default as IfViewport } from './IfViewport';
export { default as createComponent } from './createComponent';
