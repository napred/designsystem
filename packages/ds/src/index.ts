export * from './cache';
export * from './css';
export * from './hooks';
export * from './styles';
export * from './system';
export * from './theme';
export * from './types';
export * from './utilities';
export { default as css } from './css';
export { default as DesignSystem } from './DesignSystem';
export { default as DesignSystemContext } from './context';
export { default as IfViewport } from './IfViewport';
export { default as createComponent } from './createComponent';
