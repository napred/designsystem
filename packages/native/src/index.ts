export * from './createStyleApplicator';
export * from './DesignSystem';
export * from './types';

export { default as createComponent } from './createComponent';
export { default as createStyle } from './createStyle';
export { default as defaultTheme } from './defaultTheme';
export { default as DesignSystem } from './DesignSystem';

export * from '@napred/ds';
