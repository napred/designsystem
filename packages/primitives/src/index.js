// @flow

export { default as Box } from './Box';
export { default as Card, CardFooter, CardHeader } from './Card';
export { default as Flex } from './Flex';
export { default as Image } from './Image';
export { default as Link } from './Link';
export { default as Text } from './Text';
export { default as Title } from './Title';
