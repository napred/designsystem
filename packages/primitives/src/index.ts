import { createComponent } from '@napred/browser';
import * as BoxDef from './Box';
import * as CardDef from './Card';
import * as FlexDef from './Flex';
import * as ImageDef from './Image';
import * as LinkDef from './Link';
import * as TextDef from './Text';
import * as TitleDef from './Title';

export const Box = createComponent(BoxDef.name, 'div');

Box.defaultProps = BoxDef.defaultProps;

export const Card = createComponent(CardDef.Card.name, 'div');
export const CardFooter = createComponent(CardDef.CardFooter.name, 'div');
export const CardHeader = createComponent(CardDef.CardHeader.name, 'div');

Card.defaultProps = CardDef.Card.defaultProps;
CardFooter.defaultProps = CardDef.CardFooter.defaultProps;
CardHeader.defaultProps = CardDef.CardHeader.defaultProps;

export const Flex = createComponent(FlexDef.name, 'div');

Flex.defaultProps = FlexDef.defaultProps;

export const Image = createComponent(ImageDef.name, 'img');

Image.defaultProps = ImageDef.defaultProps;

export const Link = createComponent(LinkDef.name, 'a', {
  style: LinkDef.style,
});

Link.defaultProps = LinkDef.defaultProps;

export const Text = createComponent(TextDef.name, 'div');

Text.defaultProps = TextDef.defaultProps;

export const Title = createComponent(TitleDef.name, 'h1');

Title.defaultProps = TitleDef.defaultProps;
