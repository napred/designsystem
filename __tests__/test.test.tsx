/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from 'react-testing-library';
import { DesignSystem } from '../packages/ds/src';
import { Link } from '../packages/primitives/src';
import { Button } from '../packages/ui/src';

describe('test', () => {
  it('works correctly', () => {
    const { asFragment } = render(
      <DesignSystem>
        <Button as={Link} disabled variant="primary">
          Trololo
        </Button>
      </DesignSystem>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
