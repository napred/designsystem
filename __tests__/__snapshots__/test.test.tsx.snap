// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test works correctly 1`] = `
<DocumentFragment>
  .css-fisdg {
  outline: none;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.6s ease;
  transition: all 0.6s ease;
  overflow: hidden;
  cursor: not-allowed;
  -webkit-filter: grayscale(10%);
  filter: grayscale(10%);
  opacity: 0.6;
  background-color: #D03C70;
  color: #F7F7F7;
  border-color: transparent;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.css-fisdg:focus {
  outline: 0;
}

.css-t4gf00 {
  border-radius: 5px;
  border-style: solid;
  border-width: 1px;
  color: #A8A8A8;
  display: inline-block;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  font-weight: 600;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  position: relative;
}

<a
    class="css-fisdg css-t4gf00"
    disabled=""
  >
    Trololo
  </a>
</DocumentFragment>
`;
